### ISSUE TITLE
Strange gun bolt behavior and crash

### ISSUE THREAD

**Issue by Errant-4** [space-wizards/space-station-14#19165]
## Description
Closing the bolt on a spawned full gun for some reason ejects the cartridge in the chamber, when they normally eject when the bolt is opened

The bolt can also crash the client

**Reproduction**
Spawn any gun with a bolt.
Remove magazine.
Close the bolt. 
.  Cartridge is ejected from the chamber. Why?
Open bolt
Put cartridge back into gun
Close the bolt. 
Put magazine into gun.
Open and close bolt.
.  Client crashes.

**Comment by metalgearsloth** [comment]
trace?

**Comment by Errant-4** [comment]
Sorry

```
[ERRO] root: Predicting the deletion of a networked entity: cartridge (.30 rifle) (1757, CartridgeLightRifle). Trace:    at System.Environment.get_StackTrace()
   at Robust.Client.GameStates.ClientDirtySystem.OnTerminate(EntityTerminatingEvent& ev) in /home/useriel/s14/RobustToolbox/Robust.Client/GameStates/ClientDirtySystem.cs:line 47
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass6_0`1.<SubscribeEvent>b__0(Unit& ev) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Broadcast.cs:line 191
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEventCore(EventSource source, Unit& unitRef, EventData subs, Boolean byRef) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Broadcast.cs:line 332
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast, Boolean byRef) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 226
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent& args, Boolean broadcast) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 200
   at Robust.Shared.GameObjects.EntityManager.RecursiveFlagEntityTermination(EntityUid uid, MetaDataComponent metadata, EntityQuery`1 metaQuery, EntityQuery`1 xformQuery) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.cs:line 536
   at Robust.Shared.GameObjects.EntityManager.DeleteEntity(EntityUid e) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.cs:line 518
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.CycleCartridge(EntityUid uid, ChamberMagazineAmmoProviderComponent component, Nullable`1 user, AppearanceComponent appearance) in /home/useriel/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 226
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.SetBoltClosed(EntityUid uid, ChamberMagazineAmmoProviderComponent component, Boolean value, Nullable`1 user, AppearanceComponent appearance, ItemSlotsComponent slots) in /home/useriel/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 147
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.ToggleBolt(EntityUid uid, ChamberMagazineAmmoProviderComponent component, Nullable`1 user) in /home/useriel/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 242
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.OnChamberActivate(EntityUid uid, ChamberMagazineAmmoProviderComponent component, ActivateInWorldEvent args) in /home/useriel/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 68
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass47_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 255
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass58_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 393
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass67_1.<EntCollectOrdered>b__0(Unit& ev) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 572
   at Robust.Shared.GameObjects.EntityEventBus.DispatchOrderedEvents(Unit& eventArgs, ValueList`1& found) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Ordering.cs:line 54
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalOrdered(EntityUid uid, Type eventType, EventData subs, Unit& unitRef, Boolean broadcast, Boolean byRef) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Ordering.cs:line 45
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast, Boolean byRef) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 218
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 182
   at Robust.Shared.GameObjects.EntitySystem.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntitySystem.cs:line 150
   at Content.Shared.Interaction.SharedInteractionSystem.InteractionActivate(EntityUid user, EntityUid used, Boolean checkCanInteract, Boolean checkUseDelay, Boolean checkAccess) in /home/useriel/s14/Content.Shared/Interaction/SharedInteractionSystem.cs:line 934
   at Content.Shared.Hands.EntitySystems.SharedHandsSystem.TryActivateItemInHand(EntityUid uid, HandsComponent handsComp, String handName) in /home/useriel/s14/Content.Shared/Hands/EntitySystems/SharedHandsSystem.Interactions.cs:line 114
   at Content.Shared.Hands.EntitySystems.SharedHandsSystem.HandleActivateItemInHand(RequestActivateInHandEvent msg, EntitySessionEventArgs args) in /home/useriel/s14/Content.Shared/Hands/EntitySystems/SharedHandsSystem.Interactions.cs:line 60
   at Robust.Shared.GameObjects.EventBusExt.HandlerWrapper`1.Invoke(EntitySessionMessage`1 msg) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EventBusExt.cs:line 46
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass4_0`1.<SubscribeEvent>b__0(Unit& ev) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Broadcast.cs:line 169
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEventCore(EventSource source, Unit& unitRef, EventData subs, Boolean byRef) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Broadcast.cs:line 332
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEvent(EventSource source, Unit& unitRef, Type eventType, Boolean byRef) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Broadcast.cs:line 317
   at Robust.Shared.GameObjects.EntityEventBus.RaiseEvent[T](EventSource source, T toRaise) in /home/useriel/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Broadcast.cs:line 271
   at Robust.Client.GameObjects.ClientEntityManager.RaisePredictiveEvent[T](T msg) in /home/useriel/s14/RobustToolbox/Robust.Client/GameObjects/ClientEntityManager.cs:line 113
   at Content.Client.Hands.Systems.HandsSystem.UIHandActivate(String handName) in /home/useriel/s14/Content.Client/Hands/Systems/HandsSystem.cs:line 230
   at Content.Client.UserInterface.Systems.Hands.HandsUIController.HandPressed(GUIBoundKeyEventArgs args, SlotControl hand) in /home/useriel/s14/Content.Client/UserInterface/Systems/Hands/HandsUIController.cs:line 83
   at Content.Client.UserInterface.Controls.SlotControl.OnButtonPressed(GUIBoundKeyEventArgs args) in /home/useriel/s14/Content.Client/UserInterface/Controls/SlotControl.cs:line 203
   at Robust.Client.UserInterface.Control.KeyBindDown(GUIBoundKeyEventArgs args) in /home/useriel/s14/RobustToolbox/Robust.Client/UserInterface/Control.Input.cs:line 41
   at Robust.Client.UserInterface.UserInterfaceManager.<>c.<KeyBindDown>b__124_0(Control c, GUIBoundKeyEventArgs ev) in /home/useriel/s14/RobustToolbox/Robust.Client/UserInterface/UserInterfaceManager.Input.cs:line 105
   at Robust.Client.UserInterface.UserInterfaceManager._doGuiInput(Control control, GUIBoundKeyEventArgs guiEvent, Action`2 action, Boolean ignoreStop) in /home/useriel/s14/RobustToolbox/Robust.Client/UserInterface/UserInterfaceManager.Input.cs:line 279
   at Robust.Client.UserInterface.UserInterfaceManager.KeyBindDown(BoundKeyEventArgs args) in /home/useriel/s14/RobustToolbox/Robust.Client/UserInterface/UserInterfaceManager.Input.cs:line 105
   at Robust.Client.UserInterface.UserInterfaceManager.OnUIKeyBindStateChanged(BoundKeyEventArgs args) in /home/useriel/s14/RobustToolbox/Robust.Client/UserInterface/UserInterfaceManager.Input.cs:line 433
   at Robust.Client.Input.InputManager.SetBindState(KeyBinding binding, BoundKeyState state, Boolean uiOnly) in /home/useriel/s14/RobustToolbox/Robust.Client/Input/InputManager.cs:line 404
   at Robust.Client.Input.InputManager.DownBind(KeyBinding binding, Boolean uiOnly, Boolean isRepeat) in /home/useriel/s14/RobustToolbox/Robust.Client/Input/InputManager.cs:line 355
   at Robust.Client.Input.InputManager.KeyDown(KeyEventArgs args) in /home/useriel/s14/RobustToolbox/Robust.Client/Input/InputManager.cs:line 276
   at Robust.Client.GameController.KeyDown(KeyEventArgs keyEvent) in /home/useriel/s14/RobustToolbox/Robust.Client/GameController/GameController.Input.cs:line 12
   at Robust.Client.Graphics.Clyde.Clyde.DispatchSingleEvent(DEventBase ev) in /home/useriel/s14/RobustToolbox/Robust.Client/Graphics/Clyde/Clyde.Events.cs:line 47
   at Robust.Client.Graphics.Clyde.Clyde.DispatchEvents() in /home/useriel/s14/RobustToolbox/Robust.Client/Graphics/Clyde/Clyde.Events.cs:line 31
   at Robust.Client.Graphics.Clyde.Clyde.ProcessInput(FrameEventArgs frameEventArgs) in /home/useriel/s14/RobustToolbox/Robust.Client/Graphics/Clyde/Clyde.Windowing.cs:line 392
   at Robust.Client.GameController.Input(FrameEventArgs frameEventArgs) in /home/useriel/s14/RobustToolbox/Robust.Client/GameController/GameController.cs:line 559
   at Robust.Client.GameController.<StartupContinue>b__57_2(Object sender, FrameEventArgs args) in /home/useriel/s14/RobustToolbox/Robust.Client/GameController/GameController.cs:line 237
   at Robust.Shared.Timing.GameLoop.Run() in /home/useriel/s14/RobustToolbox/Robust.Shared/Timing/GameLoop.cs:line 184
   at Robust.Client.GameController.ContinueStartupAndLoop(DisplayMode mode) in /home/useriel/s14/RobustToolbox/Robust.Client/GameController/GameController.Standalone.cs:line 134
   at Robust.Client.GameController.GameThreadMain(DisplayMode mode) in /home/useriel/s14/RobustToolbox/Robust.Client/GameController/GameController.Standalone.cs:line 119
   at Robust.Client.GameController.<>c__DisplayClass100_0.<Run>b__0() in /home/useriel/s14/RobustToolbox/Robust.Client/GameController/GameController.Standalone.cs:line 86
```
An interesting thing I just noticed that the initial round does not ALWAYS eject on closing the bolt the first time. Right now it did not, and then the gun did not crash on the toggling of the bolt when going through these exact steps, I could cycle out multiple bullets. But then if I removed the magazine, put every round back where it belongs, and put back the magazine and cycled, it crashed as expected

**Comment by Errant-4** [comment]
I can no longer cause a crash as described above. However, I can still cause a (different) exception like this:

1. Spawn on devmap
2. Remove the magazine from the AKMS
3. Cycle a round out of the magazine 
4. Put the round back in the magazine
5. Put the magazine in the AKMS 
6. Close the bolt

```
Exception has occurred: CLR/System.ArgumentException
An unhandled exception of type 'System.ArgumentException' occurred in Robust.Shared.dll: 'Entity 1753 is not valid.'
   at Robust.Shared.GameObjects.EntityManager.AddComponent[T](EntityUid uid, T component, Boolean overwrite) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.Components.cs:line 258
   at Robust.Shared.GameObjects.EntityManager.AddComponent[T](EntityUid uid) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityManager.Components.cs:line 197
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.OnBallisticTakeAmmo(EntityUid uid, BallisticAmmoProviderComponent component, TakeAmmoEvent args) in /home/user/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs:line 219
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass47_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 255
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass58_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 393
   at Robust.Shared.GameObjects.EntityEventBus.EntDispatch(EntityUid euid, Type eventType, Unit& args, Boolean dispatchByReference) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 554
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast, Boolean byRef) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 222
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 182
   at Robust.Shared.GameObjects.EntitySystem.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntitySystem.cs:line 150
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.CycleCartridge(EntityUid uid, ChamberMagazineAmmoProviderComponent component, Nullable`1 user, AppearanceComponent appearance) in /home/user/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 210
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.SetBoltClosed(EntityUid uid, ChamberMagazineAmmoProviderComponent component, Boolean value, Nullable`1 user, AppearanceComponent appearance, ItemSlotsComponent slots) in /home/user/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 147
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.ToggleBolt(EntityUid uid, ChamberMagazineAmmoProviderComponent component, Nullable`1 user) in /home/user/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 242
   at Content.Shared.Weapons.Ranged.Systems.SharedGunSystem.OnChamberActivate(EntityUid uid, ChamberMagazineAmmoProviderComponent component, ActivateInWorldEvent args) in /home/user/s14/Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.ChamberMagazine.cs:line 68
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass47_0`2.<SubscribeLocalEvent>g__EventHandler|0(EntityUid uid, IComponent comp, TEvent& args) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 255
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass58_0`1.<EntSubscribe>b__0(EntityUid uid, IComponent comp, Unit& ev) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 393
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass67_1.<EntCollectOrdered>b__0(Unit& ev) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 572
   at Robust.Shared.GameObjects.EntityEventBus.DispatchOrderedEvents(Unit& eventArgs, ValueList`1& found) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Ordering.cs:line 54
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalOrdered(EntityUid uid, Type eventType, EventData subs, Unit& unitRef, Boolean broadcast, Boolean byRef) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Ordering.cs:line 45
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEventCore(EntityUid uid, Unit& unitRef, Type type, Boolean broadcast, Boolean byRef) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 218
   at Robust.Shared.GameObjects.EntityEventBus.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntityEventBus.Directed.cs:line 182
   at Robust.Shared.GameObjects.EntitySystem.RaiseLocalEvent[TEvent](EntityUid uid, TEvent args, Boolean broadcast) in /home/user/s14/RobustToolbox/Robust.Shared/GameObjects/EntitySystem.cs:line 150
   at Content.Shared.Interaction.SharedInteractionSystem.InteractionActivate(EntityUid user, EntityUid used, Boolean checkCanInteract, Boolean checkUseDelay, Boolean checkAccess) in /home/user/s14/Content.Shared/Interaction/SharedInteractionSystem.cs:line 934
   at Content.Shared.Interaction.SharedInteractionSystem.HandleActivateItemInWorld(ICommonSession session, EntityCoordinates coords, EntityUid uid) in /home/user/s14/Content.Shared/Interaction/SharedInteractionSystem.cs:line 894
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.<>c__DisplayClass5_0.<.ctor>b__0(PointerInputCmdArgs& args) in /home/user/s14/RobustToolbox/Robust.Shared/Input/Binding/InputCmdHandler.cs:line 99
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.HandleCmdMessage(ICommonSession session, InputCmdMessage message) in /home/user/s14/RobustToolbox/Robust.Shared/Input/Binding/InputCmdHandler.cs:line 119
   at Robust.Client.GameObjects.InputSystem.PredictInputCommand(FullInputCmdMessage inputCmd) in /home/user/s14/RobustToolbox/Robust.Client/GameObjects/EntitySystems/InputSystem.cs:line 106
   at Robust.Client.GameStates.ClientGameStateManager.PredictTicks(GameTick predictionTarget) in /home/user/s14/RobustToolbox/Robust.Client/GameStates/ClientGameStateManager.cs:line 433
   at Robust.Client.GameStates.ClientGameStateManager.ApplyGameState() in /home/user/s14/RobustToolbox/Robust.Client/GameStates/ClientGameStateManager.cs:line 386
   at Robust.Client.GameController.Tick(FrameEventArgs frameEventArgs) in /home/user/s14/RobustToolbox/Robust.Client/GameController/GameController.cs:line 605
   at Robust.Client.GameController.<StartupContinue>b__57_0(Object sender, FrameEventArgs args) in /home/user/s14/RobustToolbox/Robust.Client/GameController/GameController.cs:line 221
   at Robust.Shared.Timing.GameLoop.Run() in /home/user/s14/RobustToolbox/Robust.Shared/Timing/GameLoop.cs:line 233
   at Robust.Client.GameController.ContinueStartupAndLoop(DisplayMode mode) in /home/user/s14/RobustToolbox/Robust.Client/GameController/GameController.Standalone.cs:line 134
   at Robust.Client.GameController.GameThreadMain(DisplayMode mode) in /home/user/s14/RobustToolbox/Robust.Client/GameController/GameController.Standalone.cs:line 119
   at Robust.Client.GameController.<>c__DisplayClass100_0.<Run>b__0() in /home/user/s14/RobustToolbox/Robust.Client/GameController/GameController.Standalone.cs:line 86
```

**Comment by Errant-4** [comment]
Fixed by #24283
